pkg/controller/elasticsearch/validation/autoscaling_validation.go

```diff
@@ -0,0 +1,281 @@
+					"ML nodes must be in a dedicated autoscaling policy"),
+			)
+		}
+
+		if !(autoscalingSpec.NodeCount.Min >= 0) {
+			errs = append(
+				errs,
+				field.Invalid(
+					autoscalingSpecPath(i, "resources", "nodeCount", "min"),
+					autoscalingSpec.NodeCount.Min,
+					"min count must be equal or greater than 0",
+				),
+			)
+		}
+
+		if !(autoscalingSpec.NodeCount.Max > 0) {
+			errs = append(
+				errs,
+				field.Invalid(
+					autoscalingSpecPath(i, "resources", "nodeCount", "max"),
+					autoscalingSpec.NodeCount.Max,
+					"max count must be greater than 0"),
+			)
+		}
+
+		if !(autoscalingSpec.NodeCount.Max >= autoscalingSpec.NodeCount.Min) {
+			errs = append(
+				errs,
+				field.Invalid(autoscalingSpecPath(i, "resources", "nodeCount", "max"),
+					autoscalingSpec.NodeCount.Max,
+					"max node count must be an integer greater or equal than the min node count"),
+			)
+		}
+
+		// Validate CPU
+		errs = validateQuantities(errs, autoscalingSpec.CPU, i, "cpu", minCPU)
+
+		// Validate Memory
+		errs = validateQuantities(errs, autoscalingSpec.Memory, i, "memory", minMemory)
+
+		// Validate storage
+		errs = validateQuantities(errs, autoscalingSpec.Storage, i, "storage", minStorage)
+	}
+	return errs
+}
+
+// autoscalingSpecPath helps to compute the path used in validation error fields.
+func autoscalingSpecPath(index int, child string, moreChildren ...string) *field.Path {
+	return field.NewPath("metadata").
+		Child("annotations", `"`+esv1.ElasticsearchAutoscalingSpecAnnotationName+`"`).
+		Index(index).
+		Child(child, moreChildren...)
+}
+
+// validateQuantities ensures that a quantity range is valid.
+func validateQuantities(
+	errs field.ErrorList,
+	quantityRange *esv1.QuantityRange,
+	index int,
+	resource string,
+	minQuantity resource.Quantity,
+) field.ErrorList {
+	var quantityErrs field.ErrorList
+	if quantityRange == nil {
+		return errs
+	}
+
+	if !minQuantity.IsZero() && !(quantityRange.Min.Cmp(minQuantity) >= 0) {
+		quantityErrs = append(
+			quantityErrs,
+			field.Required(
+				autoscalingSpecPath(index, "minAllowed", resource),
+				fmt.Sprintf("min quantity must be greater than %s", minQuantity.String())),
+		)
+	}
+
+	// A quantity must always be greater than 0.
+	if minQuantity.IsZero() && !(quantityRange.Min.Value() > 0) {
+		quantityErrs = append(
+			quantityErrs,
+			field.Required(
+				autoscalingSpecPath(index, "minAllowed", resource),
+				"min quantity must be greater than 0"),
+		)
+	}
+
+	if quantityRange.Min.Cmp(quantityRange.Max) > 0 {
+		quantityErrs = append(
+			quantityErrs,
+			field.Invalid(
+				autoscalingSpecPath(index, "maxAllowed", resource), quantityRange.Max.String(),
+				"max quantity must be greater or equal than min quantity"),
+		)
+	}
+	return append(errs, quantityErrs...)
+}
+
+// containsStringSlice returns true if an ordered slice is included in a slice of ordered slices.
+func containsStringSlice(slices [][]string, slice []string) bool {
```

Comment:
Do we care about order here? Maybe this could be an `Intersect(o StringSet)` function on `StringSet` instead? 

```Go
len(set1.Intersect(set2)) == len(set1)
```

or simply

```Go
reflect.DeepEquals(set1, set2)
```